src/Plugin/CustomerData/UpdateCart.php

```diff
@@ -0,0 +1,76 @@
+<?php
+declare(strict_types=1);
+
+namespace Hyva\Checkout\Plugin\CustomerData;
+
+use Magento\Checkout\CustomerData\Cart;
+use Magento\Checkout\Model\Session as CheckoutSession;
+use Magento\Quote\Model\QuoteIdToMaskedQuoteIdInterface;
+
+/**
+ * Plugin to update the cart info in customerData that passes to the frontend.
+ *
+ * It plugin to the class: Magento\Checkout\CustomerData\Cart
+ */
+class UpdateCart
+{
+    /**
+     * @var \Magento\Quote\Model\Quote|null
+     */
+    private $quote = null;
+
+    /**
+     * @var \Magento\Checkout\Model\Session
+     */
+    private $checkoutSession;
+
+    /**
+     * @var \Magento\Quote\Model\QuoteIdToMaskedQuoteIdInterface
+     */
+    private $quoteIdToMaskedQuote;
+
+    /**
+     * UpdateCart constructor.
+     *
+     * @param \Magento\Checkout\Model\Session $checkoutSession
+     * @param \Magento\Quote\Model\QuoteIdToMaskedQuoteIdInterface $quoteToMaskedQuote
+     */
+    public function __construct(CheckoutSession $checkoutSession, QuoteIdToMaskedQuoteIdInterface $quoteToMaskedQuote)
+    {
+        $this->checkoutSession = $checkoutSession;
+        $this->quoteIdToMaskedQuote = $quoteToMaskedQuote;
+    }
+
+    /**
+     * Appending cart mask id into the result
+     *
+     * @param \Magento\Checkout\CustomerData\Cart $cartCustomerData
+     * @param array $result
+     * @return array
+     */
+    public function afterGetSectionData(Cart $cartCustomerData, array $result)
+    {
+        try {
+            $quoteId = (int)$this->getQuote()->getId();
+            $maskId = $this->quoteIdToMaskedQuote->execute($quoteId);
+            $result['cartId'] = $maskId;
```

Comment:
I noticed on demo.hyva.io that `cartId` is already included (although it is an empty string if I don't have a cart yet): 


![Screenshot_20210504_125421](https://user-images.githubusercontent.com/367320/116993592-fbf4c280-acd7-11eb-9de7-ee79644ea73e.png)

So this would overwrite the existing value. Is it really necessary? Or maybe only if the checkout is not used together with the Hyvä theme module?

Reply:
As discussed, we'll add a dependency to https://github.com/hyva-themes/magento2-graphql-tokens instead :)